src/test/java/uk/gov/pay/api/service/GetPaymentServiceLedgerTest.java

```diff
@@ -0,0 +1,174 @@
+import uk.gov.pay.api.auth.Account;
+import uk.gov.pay.api.exception.GetChargeException;
+import uk.gov.pay.api.ledger.service.LedgerUriGenerator;
+import uk.gov.pay.api.model.CardPayment;
+import uk.gov.pay.api.model.PaymentState;
+import uk.gov.pay.api.model.TokenPaymentType;
+import uk.gov.pay.api.model.links.PaymentWithAllLinks;
+import uk.gov.pay.api.model.links.PostLink;
+import uk.gov.pay.api.utils.mocks.ConnectorMockClient;
+import uk.gov.pay.commons.model.SupportedLanguage;
+import uk.gov.pay.commons.testing.pact.consumers.PactProviderRule;
+import uk.gov.pay.commons.testing.pact.consumers.Pacts;
+
+import javax.ws.rs.client.Client;
+import java.util.Collections;
+import java.util.Optional;
+
+import static org.hamcrest.Matchers.containsString;
+import static org.hamcrest.Matchers.notNullValue;
+import static org.hamcrest.Matchers.nullValue;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertThat;
+import static org.mockito.Mockito.when;
+import static uk.gov.pay.commons.testing.port.PortFactory.findFreePort;
+
+@RunWith(MockitoJUnitRunner.class)
+public class GetPaymentServiceLedgerTest {
+
+    private static final String ACCOUNT_ID = "123456";
+    private static final String CHARGE_ID_NON_EXISTENT_IN_CONNECTOR = "ch_123abc456xyz";
+    private static final int CONNECTOR_PORT = findFreePort();
+
+    private GetPaymentService getPaymentService;
+
+    @ClassRule
+    public static WireMockClassRule connectorRule = new WireMockClassRule(CONNECTOR_PORT);
+
+    @Rule
+    public PactProviderRule ledgerRule = new PactProviderRule("ledger", this);
+
+    @Mock
+    private PublicApiConfig mockConfiguration;
+
+    @Before
+    public void setup() {
+        connectorRule.resetAll();
+        when(mockConfiguration.getConnectorUrl()).thenReturn("http://localhost:" + CONNECTOR_PORT);
+        when(mockConfiguration.getLedgerUrl()).thenReturn(ledgerRule.getUrl());
+        when(mockConfiguration.getUseLedgerForGetPayment()).thenReturn(true);
+        when(mockConfiguration.getBaseUrl()).thenReturn("http://publicapi.test.localhost/");
+
+        ConnectorMockClient connectorMockClient = new ConnectorMockClient(connectorRule);
+        connectorMockClient.respondChargeNotFound(ACCOUNT_ID, CHARGE_ID_NON_EXISTENT_IN_CONNECTOR, "not found");
+
+        PublicApiUriGenerator publicApiUriGenerator = new PublicApiUriGenerator(mockConfiguration);
+        ConnectorUriGenerator connectorUriGenerator = new ConnectorUriGenerator(mockConfiguration);
+        LedgerUriGenerator ledgerUriGenerator = new LedgerUriGenerator(mockConfiguration);
+        Client client = RestClientFactory.buildClient(new RestClientConfig(false));
+        getPaymentService = new GetPaymentService(client, publicApiUriGenerator, connectorUriGenerator, ledgerUriGenerator, mockConfiguration);
+    }
+
+    @Test
+    @PactVerification({"ledger"})
+    @Pacts(pacts = {"publicapi-ledger-get-payment-with-metadata"})
```

Comment:
I realise you are duplicating existing patterns, but I think we need to rethink this pattern of creating a new pact for every variant of thing that can be returned from connector. We should probably just have one test for the maximal set of all attributes and leave it at that unless there is a compelling reason not to

Reply:
yep - was considering min vs max set of attributes, but thought it will be easier to leave as is for the benefit of "connector" compatibility.